### PR TITLE
Move getCurrentLanguage() to JLanguageHelper

### PR DESCRIPTION
**Pull request by andrepereiradasilva**
##### Description

This PR moves `JHelper->getCurrentLanguage()` and `JHelperContent::getCurrentLanguage()` to `JLanguageHelper::getCurrentLanguage()`, deprecates the old methods and replaces them where they are currently being used.
##### How to test
1. Use a joomla multilanguage site and apply this patch
2. Add tags in several languages
3. Add those tags to a content
4. Go to tags component (com_tags) options and in "Item selection" -> "Language Filter" select "Current"
5. Go to frontend and check that the tags that are displayed in the content are only those in the current language and the ones with all languages
6. Check also the same behaviour for the tags in the modules (mod_tags_popular and mod_tags_similar)

In conclusion, all should work as before.

@joomdonation here you have it.

@mbabker if ok for you, IMHO this should also be done in joomla framework (language).

### CODE REVIEW

**Comment by mbabker** [comment]
This isn't a change suitable for the Framework in its current form.  Mainly, the Framework doesn't have the notion of components nor does it mandate a use of Joomla's application and input APIs.

**Comment by andrepereiradasilva** [comment]
Ok, i see. There must be total separation in the framework, of course.

I noticed now that the joomla language framework (https://github.com/joomla-framework/language/tree/master/src) doesn't even have an helper class.

**Comment by andrepereiradasilva** [comment]
closed because lack of interrest
